### ISSUE TITLE
cleanup

### ISSUE THREAD

**Issue by yramachers** [SuperNEMO-DBD/SuperNEMOValidationTool#20]
The simplest run of the reconstruction module, followed by the ValidationParser gives me:

./build/ValidationParser -i ../ReconstructionValidationModule/Validation.root 
Processing ../ReconstructionValidationModule/Validation.root
No config file provided - using default settings
Error in <TFile::TFile>: file name is not specified
WARNING: No valid reference ROOT file given. To generate comparison plots, provide a valid reference ROOT file.

As warning and errors before proceeding fine with one exception

Plotting v_all_track_hit_counts:
Unknown variable type v_all_track_hit_counts: ignoring this branch

So, all defaults, i.e. no configs set neither in the recon module nor for the Parser.

I think the TFile Error follows from line 139 in ValidationParser.cxx when opening the refFile before checking the existence of the file name. Maybe change that.

I also notice that the README promises a .txt file with statistics but I only get the .root file and plenty (19) of .png pictures in the new plots_Validation folder.

In summary, all working fine so far but a little cleanup would be useful. Next step is to look at the content, then a new feature to allow for a TChain object instead of a TFile to get enough statistics for the validations. I've done that for your Sensitivity script for my students and it was very straightforward, just needs to read a text file of file names to be bound in a TChain inside ValidationParser.

**Comment by cherylepatrick** [comment]
Re: v_all_track_hit_counts
I meant to add an extra branch type for vectors, but didn't get round to it yet. However I added the  branch name into the tuple in readiness (and figured I would use it to test what happens if you have an invalid branch name). So that one is correct behaviour from the point of view of the ValidationParser, but if other people are going to be using my Reconstruction validation module, I should probably take it out of there.

You're right about the TFile error, I've fixed it in PR #21.

The txt file with statistics is only if you give it a reference file: updated the README to make that clear. I can add some more (non-comparison) stats easily enough, if anyone wants to give opinions on what they'd like to see in there. But for now, it's just the comparisons.

TChain is a good idea and a relatively easy add, but I'll leave that for right now as it's not strictly a bug, and I really should be working on some other stuff today. Might get that in for collaboration meeting, might not!

**Comment by yramachers** [comment]
Thanks, that clarifies it. The TChain is more a feature request than an issue. I simply wanted to flag it up lest I forget about it. I think it will be vital in order to have high statistics reference file creation at some point. The statistics output should also or mostly consider output that is suitable for automated validation by computer rather than people. We are good at looking at pictures while automation needs some numbers.
